src/Xamarin.Android.Build.Tasks/MSBuild/Xamarin/Android/Xamarin.Android.Resource.Designer.targets

```diff
@@ -100,9 +101,15 @@ Copyright (C) 2016 Xamarin. All rights reserved.
   <!-- Creates a Resource.designer.cs file in the Intermediate output path which derives from the
        Designer Assembly, for backward compatability.
    -->
+  <AndroidError Code="XA1038"
+      ResourceName="XA1038"
+      FormatArguments="AndroidResourceDesignerClassModifier;$(AndroidResourceDesignerClassModifier);&apos;public&apos;, &apos;internal&apos;"
+      Condition=" '$(AndroidResourceDesignerClassModifier)' != 'public' And '$(AndroidResourceDesignerClassModifier)' != 'internal' "
```

Comment:
One potential issue with this is that string comparisons are case *insensitive* in MSBuild.  Which means:

```
dotnet build -p:AndroidResourceDesignerClassModifier=PUBLIC
```

will be valid for this `Condition`, but result in `GenerateResourceDesignerIntermediateClass.Modifier` being set to `PUBLIC`, which will result in invalid C#.

It would likely be better if `$(AndroidResourceDesignerClassModifier)` validation were instead done within `<GenerateResourceDesignerIntermediateClass/>`, *or* if this used [MSBuild Property Functions](https://learn.microsoft.com/en-us/visualstudio/msbuild/property-functions?view=vs-2022#static-property-functions) against [`string.CompareOrdinal(string, string)`](https://learn.microsoft.com/en-us/dotnet/api/system.string.compareordinal?view=netstandard-2.0#system-string-compareordinal(system-string-system-string)), a'la (untested):

```xml
Condition="
     '$([System.String]::CompareOrdinal('public', $(AndroidResourceDesignerClassModifier))' != '0' And
     '$([System.String]::CompareOrdinal('internal', $(AndroidResourceDesignerClassModifier))' != '0'
"
```

Reply:
I'll try to use the PRoperty Functions